diesel_infer_schema/src/pg.rs

```diff
@@ -0,0 +1,22 @@
+use std::error::Error;
+
+use data_structures::*;
+
+pub fn determine_column_type(attr: &ColumnInformation) -> Result<ColumnType, Box<Error>> {
+    let is_array = attr.type_name.starts_with('_');
+    let tpe = if is_array {
+        &attr.type_name[1..]
+    } else {
+        &attr.type_name
+    };
+
+    Ok(ColumnType {
+        path: vec!["diesel".into(), "types".into(), capitalize(tpe)],
+        is_array: is_array,
+        is_nullable: attr.nullable,
+    })
+}
+
+fn capitalize(name: &str) -> String {
+    name[..1].to_uppercase() + &name[1..]
+}
```

Comment:
For symmetry, I'd add at least copy the tests `values_which_already_map_to_type_are_returned_unchanged` test from mysql and add one for detecting arrays.